### PR TITLE
CLC-5840 Distribute LTI Usage / TurnItIn reports through Google Drive

### PR DESCRIPTION
**Pull request by raydavis**
https://jira.ets.berkeley.edu/jira/browse/CLC-5840

The first commit untangles OEC-specific classes from the more generic GoogleApps module, allows SPA accounts (for example) to find shared folders, and adds a convenient upload-to-a-worksheet method.

The second commit uploads LTI and TurnItIn usage reports to a folder shared between PDG and TLS.

Bamboo testext under way, which may call for amendments.

### CODE REVIEW

File: app/models/google_apps/sheets_manager.rb [hunk 1]
```diff
@@ -97,30 +115,11 @@ def update_worksheet(worksheet, updates)
...
-            csv << csv_row
-          else
-            csv << row.values_at(*headers)
-          end
-        end
-      end
-      upload_to_spreadsheet(title, description, StringIO.new(content), parent_id)
-    end
-
-    def upload_to_spreadsheet(title, description, path_or_io, parent_id)
```
Comment (raydavis) [inline app/models/google_apps/sheets_manager.rb]: It looks as if Canvas & OEC clients both rely a lot more on worksheet titles than on Google file "descriptions". I have a hard time even finding them in the current Drive UX.

File: app/models/google_apps/sheets_manager.rb [hunk 2]
```diff
@@ -97,30 +115,11 @@ def update_worksheet(worksheet, updates)
       result.body
     end
 
-    def upload_worksheet(title, description, worksheet, parent_id = 'root', opts={})
-      content = CSV.generate do |csv|
-        headers = worksheet.headers
-        csv << headers
-        worksheet.each_sorted do |row|
```
Comment (raydavis) [inline app/models/google_apps/sheets_manager.rb]: This method is specific to Oec::Worksheet, which is a completely different animal from the GoogleDrive::Worksheet used elsewhere in this same class.

File: app/models/google_apps/sheets_manager.rb (hunk 1 above)
Comment (paulkerschen) [inline app/models/google_apps/sheets_manager.rb]: Quite so, @raydavis; to my recollection, all we were ever did with the file description was dutifully pass an empty string.

**Comment by raydavis** [comment]
Oh yeah, _definitely_ some test amendments coming up. Unfortunately I also have a meeting coming up & an emergency at home, so this PR will have to sit on ice until tomorrow.

**Pull request closed without merging**